### ISSUE TITLE
.qmd files as document templates

### ISSUE THREAD

**Issue by Robinlovelace** [rstudio/rstudio#11316]
I'm not 100% sure this is a new feature request, or even whether this is the right repo to open such an issue, but it could be cool and useful to be able to select templates for .qmd documents in the same way you can for .Rmd documents. In the below screenshot I guess my feature request is for a new option underneath 'Interactive' called "Template" or similar. This would enable/encourage others to develop their own .qmd style documents as enabled/encouraged by the rticles package (where I thought about opening this issue).

![image](https://user-images.githubusercontent.com/1825120/170824418-f84b5898-1d15-4625-b6af-84333f3dc6b0.png)

A way of creating .qmds based on templates from the command line, e.g. with rstudioapi, would also be on my wish list although that may be best handled by things external to RStudio. Apologies if you're already on the case with this and hope this request is of use/interest in any case!

**Comment by RaymondBalise** [comment]
My package `rUM` adds Rmd templates (among other things) and I would love to be able to add Quarto templates.   I just can't miss the chance to add "quart o rUM" templates in the next release of `rUM`.  So 👍  @Robinlovelace and 🙏 @ronblum.

**Comment by ronblum** [comment]
@RaymondBalise Re-reading this, if I understand correctly, you're looking for a change in Quarto using the command line interface, not specific to RStudio. Is this correct?

(I missed this the first time because I was too busy laughing at the joke. 😄 )

**Comment by cderv** [comment]
@ronblum just to share for reference that there is also a related feature request in Quarto: https://github.com/quarto-dev/quarto-cli/issues/1111

Probably syncing effort on both side will be a way to make this comes true.

**Comment by ronblum** [comment]
Thanks all! I'll raise this.

Triage: See discussion above and in [Quarto CLI request 1111](https://github.com/quarto-dev/quarto-cli/issues/1111).

**Comment by cderv** [comment]
We probably to keep that open

**Comment by cderv** [comment]
We probably want to keep that open

**Comment by RaymondBalise** [comment]
Yes please do. I would love to update my rUM package to include hooks into the IDE for Quarto. My students would certainly appreciate it.  As it stands they have to go to download gists instead of just using the UI.

**Comment by venpopov** [comment]
Please do add this feature!

**Comment by jthomasmock** [comment]
I played around with creating a template through the RMarkdown system. This requires a template file in `.Rmd` OR `.rmd` and `.qmd` will not be recognized. See: https://github.com/rstudio/rstudio/pull/9454/files

Ironically, because a `.rmd` with Quarto yaml is treated like Quarto:

```
---
format: html
---
```

A `.Rmd` _could_ be used to create a Quarto template and distribute through an R package.

**Comment by RaymondBalise** [comment]
Thanks for giving it a look Tom.  I use the R Markdown templates that I baked into {`rUM`} to help novices get started.  I _really_  want to instead teach them with Quarto from the get-go.  I think telling them to use the R Markdown templates to create a quarto document (and then to change the file suffix) would be _**bad mojo**_ for absolute novices.

Given that Quarto can now nearly completely replace R Markdown, it would be extremely good, especially for instructors, if we could use/access a Quarto template store in the RStudio IDE.

**Comment by isaactpetersen** [comment]
I just want to add my voice in support of adding the option of using Quarto templates via RStudio dialogs (as is supported with R Markdown templates). This will allow Quarto to be more accessible, including to students. Thanks!

**Comment by AaronRendahl** [comment]
I'd love to have this too!

**Comment by maldridgeepa** [comment]
I'd also love this feature. we have created R project templates to standardize work products in our group, and this includes R-markdown templates. There are good applications for us to use quarto, but it currently requires more work because our document templates are not available in quarto.
